pkg/controllers/provisioner/v1alpha1/allocation/greedyallocator.go

```diff
@@ -45,25 +46,69 @@ func (a *GreedyAllocator) Allocate(pods []*v1.Pod) error {
 
 type SchedulingGroup struct {
 	Pods        []*v1.Pod
-	Constraints cloudprovider.CapacityConstraints
+	Constraints *cloudprovider.CapacityConstraints
 }
 
-func (a *GreedyAllocator) getSchedulingGroups(pods []*v1.Pod) []SchedulingGroup {
-	groups := []SchedulingGroup{}
-
+func (a *GreedyAllocator) getSchedulingGroups(pods []*v1.Pod) []*SchedulingGroup {
+	groups := []*SchedulingGroup{}
 	for _, pod := range pods {
+		added := false
 		for _, group := range groups {
-			if a.matchesGroup(pod, group) {
-				group.Pods = append(group.Pods, pod)
+			if a.matchesGroup(group, pod) {
+				a.addToGroup(group, pod)
+				added = true
 				break
 			}
 		}
+		if added {
```

Comment:
This should be unnecessary due to your break statement.

Reply:
break will take out of the inner for loop. if I have already it added to the group, I don't want to do anything else I want to add to group.